presentation-api/cast.html

```diff
@@ -17,97 +17,86 @@ <h3>Background</h3>
-    updatePresentationAvailability(availability.value);
-    availability.onchange = function() {
-      updatePresentationAvailability(availability.value);
-    };
-  });
-});
-
-function setConnection(connection) {
-  presentationConnection = connection;
-  document.querySelector('#close').disabled = false;
-  document.querySelector('#terminate').disabled = false;
-  presentationConnection.onclose = function() {
-    ChromeSamples.log('Connection closed.');
-  };
-  presentationConnection.onterminate = function() {
-    ChromeSamples.log('Connection terminated.');
-  };
-}
+let presentationConnection;
 
 document.querySelector('#start').addEventListener('click', function() {
-  presentationRequest.start().then(function(connection) {
-    ChromeSamples.log('Connected to presentation: ' + connection.id);
-    setConnection(connection);
-  }).catch(function(error) {
-    ChromeSamples.log(error.name + ': ' + error.message);
+  log('Starting presentation request...');
+  presentationRequest.start()
+  .then(connection => {
+    log('> Connected to ' + connection.url + ', id: ' + connection.id);
+  })
+  .catch(error => {
+    log('> ' + error.name + ': ' + error.message);
   });
 });
 
-document.querySelector('#reconnect').addEventListener('click', function() {
-  var presentationId = document.querySelector('#presentationId').value.trim();
-  if (presentationId.length) {
-    presentationRequest.reconnect(presentationId).then(function(connection) {
-      ChromeSamples.log('Reconnected to presentation: ' + connection.id);
-      setConnection(connection);
-    }).catch(function(error) {
-      ChromeSamples.log(error.name + ': ' + error.message);
-    });
-  }
+presentationRequest.addEventListener('connectionavailable', function(event) {
+  presentationConnection = event.connection;
+  presentationConnection.addEventListener('close', function() {
+    log('> Connection closed.');
+  });
+  presentationConnection.addEventListener('terminate', function() {
+    log('> Connection terminated.');
+  });
 });
 
 document.querySelector('#close').addEventListener('click', function() {
-  if (presentationConnection) {
-    presentationConnection.close();
-    document.querySelector('#close').disabled = true;
-    document.querySelector('#terminate').disabled = true;
-  }
+  log('Closing connection...');
+  presentationConnection.close();
 });
 
 document.querySelector('#terminate').addEventListener('click', function() {
-  if (presentationConnection) {
-    presentationConnection.terminate();
-    document.querySelector('#close').disabled = true;
-    document.querySelector('#terminate').disabled = true;
-  }
+  log('Terminating connection...');
+  presentationConnection.terminate();
+});
+
+document.querySelector('#reconnect').addEventListener('click', () => {
+  const presentationId = document.querySelector('#presentationId').value.trim();
+
+  presentationRequest.reconnect(presentationId)
```

Comment:
reconnect() may not work for Cast URLs with the presentationConnection.id as given...

Reply:
I've removed it then. Thanks!